app/mailers/oral_history_delivery_mailer.rb

```diff
@@ -0,0 +1,70 @@
+class OralHistoryDeliveryMailer < ApplicationMailer
+
+  def oral_history_delivery_email
+    raise ArgumentError.new("Required params[:request] missing") unless request.present?
+    raise ArgumentError.new("Required patron email missing") unless request.patron_email.present?
+    mail(to: to_address, subject: subject, content_type: "text/html")
+  end
+
+  def request
+    @request ||= params[:request]
+  end
+
+  def to_address
+    request.patron_email
+  end
+
+  # Note: any value greater than 604800 will raise an exception.
+  # See https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/S3/Presigner.html
+  def how_long_urls_will_be_valid
+    1.week.to_i - 10
+  end
+
+
+  def created_at
+    request.created_at
+  end
+
+  def patron_name
+    request.patron_name
+  end
+
+  def work
+    request.work
+  end
+
+  def assets
+    work.members.order(:position).select {|x| x.is_a? Asset}
+  end
+
+  def asset_links
+    assets.map do |asset|
+      url = asset.file.url(
+        public: false,
+        expires_in: how_long_urls_will_be_valid,
+        response_content_type: asset.content_type,
+        response_content_disposition: ContentDisposition.format(
+          disposition: "inline",
+          filename: DownloadFilenameHelper.filename_for_asset(asset)
+        )
+      )
+      { text: asset.title, url: url }
+    end
```

Comment:
generating a list of hashes that then we need to coordinate what to expect in the hash is weird. How about you just have a method `download_url(asset)`, and then the actual .html.erb can just iterate through the assets itself, call `asset.ttitle` itself, and call `mailer.download_url(asset)` when needed.

Reply:
Good call. Fixed!